src/main/java/com/google/sps/data/GivrUser.java

```diff
@@ -140,9 +140,10 @@ public static GivrUser getUserByEmail(String email) {
   public static GivrUser getCurrentLoggedInUser() {
     UserService userService = UserServiceFactory.getUserService();
     boolean isUserLoggedIn = userService.isUserLoggedIn();
-    String url = userService.createLoginURL("/");
+    String url = "";
     
     if (isUserLoggedIn) {
+      url = userService.createLoginURL("/");
```

Comment:
erm I had misrecalled before; you only want to create the url if this is !isUserLoggedIn, right?

Reply:
yes. would this mean removing the generation completely?